### DISCUSSION TITLE
Is there a newbie day 0 tutorial

### DISCUSSION THREAD

**Discussion by McGr** [cariboulabs/cariboulite#98] [Q&A] [answered]
I'm a total newbie at SDR and would just like to get started.

I have my Cariboulite inserted in my Pi 4, and have GRC installed and operational on Bullseye (64bit). Antenna's are installed and the LEDs are lit.

Any tips on how do I get my first radio running? I've never used any of the SDR tools before so I'm scrambling through GRC tutorials.  Looks like I need some kind of source for the cariboulite but dont see anything in the GRC. I did read that this device is based on top of Soapy API and I do have a Soapy source (and an OSMOCOM source too) but don't see how to configure it.

Does anyone have an example radio I could load to try this board out?

Pointers would be very much appreciated.

Thanks Mike.

**Comment by OTLnoname** [comment, marked as answer]
By no means I possess the in-depth knowledge on everything surrounding SDR, not even close, but I will give it a try to get you some directions based on what I’ve setup so far, which is not much.

First and foremost, I don’t think cariboulite has gotten to the point of being a ready to use state yet. There will likely be more development on the software side.

My setup is based on a freshly installed 64bit bullseye desktop version on a pi 4 (I wanted to make sure everything works properly before switching over to a headless setup).
Before you run the official install.sh, you can add “dtparam=i2c_vc=on” to /boot/config.txt to avoid error during environment check at the end of the install script. Apparently, not all functions rely on i2c_vc to be on.

Once everything is ready, you can run the install script “./install.sh” in the cariboulite folder you pulled through git. It’s under the started and installation section on the code page. The installation script should install SoapySDRUtil and SoapySDRServer for you before compiling the drivers for cariboulite device. However, I found that SoapySDRServer never got installed properly for me so I had to compile it myself.  At this point, you can use “cariboulite_test_app” and “cariboulite_util” in the cariboulite/build folder to manually interact with cariboulite.

```
sudo ./cariboulite_test_app

#You should also be able to see cariboulite devices (HiF and S1G) using SoapySDRUtil
#the case of the letters are important
sudo SoapySDRUtil --find
```

I believe you can use the source code pulled by the script or just follow the instructions from pothosware/SoapyRemote git page as following:

```
git clone https://github.com/pothosware/SoapyRemote.git
cd SoapyRemote 
mkdir build
cd build
cmake ..
make
sudo make install 
```

Once SoapyRemote (SoapySDRServer) is compiled and installed properly, you can use SoapySDRServer to connect to it remotely using native SDR software like cubicsdr (I’ve yet to get sdrpp install properly on my pi, gqrx also has issue connecting to Soapy).

```
#To launch server manually (binding port is 55132)
sudo SoapySDRServer --bind

#Or enable the server so it starts when system boots
sudo systemctl enable SoapySDRServer
sudo systemctl start SoaypySDRServer #only for the first time, subsequent reboots do not need manual server start
```

After the server is started, you will be able to connect using cubicsdr using ip:55132. I was able to get cubicsdr, openwebrx and gqrx to see both radios on caribou using 0.0.0.0 on local host. However, gqrx seemed to have issue staying connected with SoapyRemote and cubicsdr and openwebrx are both showing poor signal strength using antennas (a 70cm/2m hand held antenna, a multi band discone and a rubber duckie) that produce adequate to good signal strength on local radio stations (~100MHz) for the RTL SDR dongle on the same pi setup, so I’m guess either I’m still missing something or the driver still needs some work. I don’t know enough to tell for sure.

This is how far I’ve gotten recently, and I hope this helps to get you started. I hope I has not made too much mistakes in write this as well.
Good luck and have fun!

**Reply by McGr** [reply]
I really appreciate your post. It got me a long way.

I also had a problem with SoapySDR not installing but this was fixed with

`sudo apt-get install soapyremote-server`

For me it looks like I still have a problem loading the SMI drivers, but I will post that as a separate question.

Thanks again, Mike

**Comment by mvmn** [comment]
Which antennas did you use for your Caribou Lite?

I presume I can use something like 915MHz antenna from FPV drone for the sub 1GHz range - though I'm not completely sure. But for 6GHz I don't have any 6GHz antennas available, and I reckon 5.8GHz from a WiFi router may not work properly - because of the standing wave ratio (SWR) mismatch etc.

Does anyone have any advices on this?
